tests/test_maintel_setupmtcs.py

```diff
@@ -0,0 +1,127 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+import logging
+import random
+import unittest
+
+from lsst.ts import salobj
+from lsst.ts import utils
+from lsst.ts.idl.enums import Script
+from lsst.ts.observatory.control.mock import MTCSMock
+from lsst.ts.standardscripts import get_scripts_dir, BaseScriptTestCase
+from lsst.ts.standardscripts.maintel import SetupMTCS
+
+random.seed(47)  # for set_random_lsst_dds_partition_prefix
+
+index_gen = utils.index_generator()
+
+# Clean up log to focus only on script
+logging.basicConfig(level=logging.DEBUG)
+logging.getLogger("asyncio").setLevel(logging.ERROR)
+logging.getLogger("lsst.ts.utils.tai").setLevel(logging.ERROR)
+logging.getLogger("MTCSMock").setLevel(logging.ERROR)
+
+
+class TestSetupMTCS(BaseScriptTestCase, unittest.IsolatedAsyncioTestCase):
+    async def basic_make_script(self, index):
+        self.script = SetupMTCS(index=index)
+        self.mtcs_mock = MTCSMock()
+        return self.script, self.mtcs_mock
+
+    async def test_configure_errors(self):
+        """Test error handling in the do_configure method."""
+        async with self.make_script():
+
+            # Check schema validation.
+            for bad_config in (
+                # The only *bad* case is that ccw_following has invalid value
+                {"ccw_following": "BADVALUE"},
+            ):
+                with self.subTest(bad_config=bad_config):
+                    with self.assertRaises(salobj.ExpectedError):
+                        await self.configure_script(**bad_config)
+
+    async def test_configure_good(self):
+        """Test configure method with valid configurations."""
+        async with self.make_script():
+
+            # Start with default configuration
+            await self.configure_script()
+
+            # The default value for `ccw_following` is True
+            self.assertEqual(self.script.config.ccw_following, True)
+
+            # Now try to change that to False
+            ccw_following = False
+            await self.configure_script(
+                ccw_following=ccw_following,
+            )
+
+            # Check configuration was correctly loaded
+            self.assertEqual(self.script.config.ccw_following, ccw_following)
+
+    async def test_executable(self):
+        scripts_dir = get_scripts_dir()
+        script_path = scripts_dir / "maintel" / "setup_mtcs.py"
+        await self.check_executable(script_path)
+
+    async def test_run(self):
+
+        # Start the test itself
+        async with self.make_script():
+
+            # ToDo - This is not implemented in the mock.
+            #  Emulating here for now.
+            class SimulationMode:
```

Comment:
Define these outside the method.. I know some tests do this but I think we should avoid it.

Reply:
Since the software version and the simulation mode are now in LOVE, I removed this from the actual code, then, we can remove this part from the test.